source/common/http/http2/codec_impl.h

```diff
@@ -100,14 +106,18 @@ class Nghttp2SessionFactory {
                     const envoy::config::core::v3::Http2ProtocolOptions& options) PURE;
 };
 
-class ProdNghttp2SessionFactory : public Nghttp2SessionFactory {
+class ProdNghttp2SessionFactory : public Http2SessionFactory {
 public:
   nghttp2_session* createOld(const nghttp2_session_callbacks* callbacks, ConnectionImpl* connection,
                              const nghttp2_option* options) override;
 
   void initOld(nghttp2_session* session, ConnectionImpl* connection,
                const envoy::config::core::v3::Http2ProtocolOptions& options) override;
 
+  std::unique_ptr<http2::adapter::Http2Adapter>
+  create(const nghttp2_session_callbacks* callbacks, ConnectionImpl* connection,
+         const http2::adapter::OgHttp2Adapter::Options& options) override;
+
   std::unique_ptr<http2::adapter::Http2Adapter> create(const nghttp2_session_callbacks* callbacks,
                                                        ConnectionImpl* connection,
                                                        const nghttp2_option* options) override;
```

Comment:
I wonder if the 2 `create` methods can be abstracted over (assuming that nghttp2 and oghttp2 will keep being 2 possible in the long run).

Reply:
In the medium term, we plan to remove nghttp2.